src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -169,6 +169,23 @@ protected void AssertSelf ()
 			throw new ObjectDisposedException (nameof (AndroidClientHandler));
 		}
 
+		string EncodeUrl (Uri url)
+		{
+			if (url == null)
+				return String.Empty;
+
+			if (String.IsNullOrEmpty (url.Query))
```

Comment:
Most URLs will likely have no query part - no reason to waste time and memory reconstructing the URL with UriBuilder